pkg/image/importer/credentials.go

```diff
@@ -161,7 +161,12 @@ func (s *SecretCredentialStore) init() credentialprovider.DockerKeyring {
 
 func basicCredentialsFromKeyring(keyring credentialprovider.DockerKeyring, target *url.URL) (string, string) {
 	// TODO: compare this logic to Docker authConfig in v2 configuration
-	value := target.Host + target.Path
+	var value string
+	if len(target.Scheme) == 0 || target.Scheme == "https" {
+		value = target.Host + target.Path
+	} else {
+		value = target.String()
```

Comment:
Why `http://` prefixed target is treated this way? It deserves at least a comment.

Reply:
We don't want to send https credentials to http endpoints.  You have to explicitly say you want to send credentials to an insecure endpoint.